plone/session/tests/testPAS.py

```diff
@@ -117,7 +117,10 @@ def testCredentialsUpdateAnonymous(self):
         session = self.folder.pas.session
         request = self.makeRequest("test string")
         session.updateCredentials(request, request.response, "our_user", "password")
-        self.assertIsNotNone(
+        # The anonymous user should not get a cookie: resetCredentials should
+        # not do anything when there are no current credentials.
+        # See https://github.com/plone/Products.CMFPlone/issues/3835
+        self.assertIsNone(
             request.response.getCookie(session.cookie_name),
         )
```

Comment:
I think this test is now equivalent to the one above it, since in both cases the user is not logged in.

Reply:
Indeed in both cases the user is not logged in.
But in this test we check that calling `updateCredentials` for a _known_ user does not set a cookie.
In the test above it, we check that calling `updateCredentials` for an _unknown_ user does not set a cookie.
So we are testing two distinct cases where `updateCredentials` should not do anything.

Reply:
I have added comments in a few tests to make clearer what we are actually testing.